alpha/lib/model/LiveEntry.php

```diff
@@ -609,19 +613,28 @@ private function storeInCache($key)
 			KalturaLog::debug("cacheStore is null. cacheType: $cacheType . returning false");
 			return false;
 		}
-		return $cacheStore->set($key, true, kConf::get('media_server_cache_expiry', 'local', self::DEFAULT_CACHE_EXPIRY));
+		KalturaLog::debug("Set cache key [$key] from store [$cacheType] ");
+		$returnValue =  $cacheStore->set($key, true, kConf::get('media_server_cache_expiry', 'local', self::DEFAULT_CACHE_EXPIRY));
+		$ans = $cacheStore->get($key);
+		KalturaLog::debug("Get cache key [$key] from store [$cacheType] returned [$ans] and reurn value was [$returnValue]");
 	}
 	
-	public function setMediaServer($index, $hostname, $applicationName = null)
+	public function setMediaServer($index, $hostname)
 	{
 		if(is_null($this->getFirstBroadcast())) 
 			$this->setFirstBroadcast(kApiCache::getTime());
-				
+
+		/* @var $mediaServerNode MediaServerNode*/
 		$mediaServerNode = ServerNodePeer::retrieveActiveMediaServerNode($hostname);
 		if (!$mediaServerNode)
 			throw new kCoreException("Media server with host name [$hostname] not found", kCoreException::MEDIA_SERVER_NOT_FOUND);
-		
-		$key = $this->getId() . "_{$hostname}_{$index}";
+
+		/* @var $dbLiveEntryServerNode LiveEntryServerNode*/
+		$dbLiveEntryServerNode = EntryServerNodePeer::retrieveByEntryIdAndServerType($this->getId(), $index);
+		if (!$dbLiveEntryServerNode)
+			throw new KalturaAPIException("Failed to find valid entry server node ", KalturaErrors::ENTRY_SERVER_NODE_NOT_FOUND);
+
+		$key = $this->getEntryServerNodeCacheKey($dbLiveEntryServerNode);
```

Comment:
Generating the same key is a overhead I don't think that we need (additional call to the DB [not sphinx] to get the hostname ) even at the cost of losing the current elements (for first deployment) that are in the cache